framework/src/base/MooseApp.C

```diff
@@ -1005,10 +1005,21 @@ MooseApp::setupOptions()
 
     if (isParamSetByUser("mesh_only"))
     {
-      _syntax.registerTaskName("mesh_only", true);
-      _syntax.addDependency("mesh_only", "setup_mesh_complete");
-      _syntax.addDependency("determine_system_type", "mesh_only");
-      _action_warehouse.setFinalTask("mesh_only");
+      // If we are looking to just check the input, there is no need to
+      // call MeshOnlyAction and generate a mesh
+      if (_check_input)
+        _action_warehouse.setFinalTask("setup_mesh_complete");
+      else
+      {
+        _syntax.registerTaskName("mesh_only", true);
+        _syntax.addDependency("mesh_only", "setup_mesh_complete");
+        _syntax.addDependency("determine_system_type", "mesh_only");
+        _action_warehouse.setFinalTask("mesh_only");
+
+        // Set these flags to enable warning messages if unused parameters exist in the input file
+        _check_input = true;
+        _enable_unused_check = WARN_UNUSED;
```

Comment:
can we do an error here?

It's standard now to error on unused parameters

Reply:
I put it as warning here because by default mesh generation happens before input checking, so the code would error but an exodus file would still be generated, which would be confusing to the user

Reply:
I think consistency primes over user confusion here. we make no guarantee not to output on errors, we should guarantee an error

if a parameter name changes in a test with `allow_warnings = true` we could never catch it